_posts/2023-01-22-auto-restarting-connectors.md

```diff
@@ -0,0 +1,328 @@
+It has also a special option `fail.task.after.records` which makes the connector tasks fail after receiving a pre-configured amount of messages.
+And we will use this feature to demonstrate the auto-restart feature.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnect
+metadata:
+  name: my-connect
+  annotations:
+    strimzi.io/use-connector-resources: "true"
+spec:
+  version: 3.3.2
+  replicas: 1
+  bootstrapServers: my-cluster-kafka-bootstrap:9092
+  config:
+    group.id: connect-cluster
+    offset.storage.topic: connect-cluster-offsets
+    config.storage.topic: connect-cluster-configs
+    status.storage.topic: connect-cluster-status
+    # -1 means it will use the default replication factor configured in the broker
+    config.storage.replication.factor: -1
+    offset.storage.replication.factor: -1
+    status.storage.replication.factor: -1
+  build:
+    output:
+      type: docker
+      image: ttl.sh/auto-restart-blog-post:latest
+    plugins:
+      - name: echo-sink-connector
+        artifacts:
+          - type: jar
+            url: https://github.com/scholzj/echo-sink/releases/download/1.3.0/echo-sink-1.3.0.jar
+            sha512sum: 7a32ab28734e4e489a2f946e379ad4266e81689d1ae1a54b6cd484ca54174eb52f271ed683f62e2fd838f48d69a847c11a6dbb4d31bf9fc5b7edd6f5463cd0b5
+```
+
+And once the Connect cluster is running, we can finally create the connector.
+Notice, that it enabled the auto-restart feature and configures the connector to have its task fail after receiving 6 messages.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnector
+metadata:
+  name: echo-sink
+  labels:
+    strimzi.io/cluster: my-connect
+spec:
+  class: EchoSink
+  tasksMax: 1
+  autoRestart:
+    enabled: true
+  config:
+    topics: "my-topic"
+    fail.task.after.records: 5
+    key.converter: org.apache.kafka.connect.storage.StringConverter
+    value.converter: org.apache.kafka.connect.storage.StringConverter
+    key.converter.schemas.enable: false
+    value.converter.schemas.enable: false
+```
+
+After we create the connector, it will create the task and it will be running without any issues because it is not receiving any messages yet.
+So we have to start a producer and send the first 6 messages.
+
+```
+kubectl run kafka-producer -ti --image=quay.io/strimzi/kafka:0.33.0-kafka-3.3.2 --rm=true --restart=Never -- bin/kafka-console-producer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic my-topic
+If you don't see a command prompt, try pressing enter.
+>Hello World 1
+>Hello World 2
+>Hello World 3
+>Hello World 4
+>Hello World 5
+>Hello World 6
+```
+
+After the sixth message, you should see the following error in the Kafka Connect log:
+
+```
+2023-01-22 23:02:52,246 WARN [echo-sink|task-0] Failing as requested after 5 records (cz.scholz.kafka.connect.echosink.EchoSinkTask) [task-thread-echo-sink-0]
```

Comment:
do we need to show all of this?

Reply:
I shortened it a bit. But I think it is valuable to have it there for the readers to see the error if they want to try it.